jaeger-production-template.yml

```diff
@@ -24,8 +24,10 @@ items:
       jaeger-infra: collector-deployment
   spec:
     replicas: 1
-    strategy:
-      type: Recreate
+    selector:
```

Comment:
* Is this selector mandatory? 
* What does it do?

Reply:
It is. Without the selector, this happens:

```
The Deployment "jaeger-collector" is invalid: 
* spec.selector: Required value
* spec.template.metadata.labels: Invalid value: map[string]string{"jaeger-infra":"collector-pod", "app":"jaeger"}: `selector` does not match template `labels`
```

It looks like it's not required when the version is set to `extensions/v1beta1`, so, I assume this became a requirement when it was moved out of beta.

Reply:
I had also a question about back compatibility. Will `apps/v1` work on older k8s versions?

Reply:
Probably not, but similarly, are we guaranteeing backwards compatibility? If so, up to which version? 

I'd rather break it "now" (if anything) that this feature moved from beta, than commit to keep backwards compat with a beta version.

Reply:
By the way: as the test framework seems to require this older notation, I'm reverting this part of the change, but the backwards compatibility question is a good one.

Reply:
I don't think we want to commit to anything right now, but k8s itself does back. compatibility for `betav1` so we could leverage it without doing anything special.

Reply:
If we can support more versions basically for free the better.